File: Recursion/FibonacciSeries/fibonacci.cpp
/*
Problem : get fibonacci nth term
*/

#include <iostream>
using namespace std;

int getFibonacci(int n){
    if(n <= 2) return 1;

    return getFibonacci(n-1) + getFibonacci(n-2);
}

int main(){

    cout << "Enter N : " ;
    int n;
    cin >> n;

    cout << getFibonacci(n) << endl;

    return 0;
}

File: Recursion/FibonacciSeries/Fibonacci.java
/*
Problem : get fibonacci nth term
*/

import java.util.Scanner;

public class Fibonacci {
    public static int getFibonacci(int n){
        if(n <= 2) return 1;

        return getFibonacci(n-1) + getFibonacci(n-2);
    }

    public static void main(String[] args) {
        System.out.println("Enter N : ");
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.close();

        System.out.println(getFibonacci(n));
    }
}
